check.py

```diff
@@ -107,6 +107,10 @@
             cmd += ['-O0'] # test that -O0 does nothing
         else:
           cmd += ['-O']
+        if 'PRINT_FULL' in asm:
```

Comment:
In that case why not have the test script only set this `if asm == 'PRINT_FULL.asm.js'`? That way the intent of multiple small tests in this one special file is documented by the code itself.